File: datasets/Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/data_preprocessing.py
# -*- coding: utf-8 -*-
"""
Created on Fri May  1 15:40:23 2020

@author: jcantero
"""
# Plantilla de Pre Procesado

# Cómo importar las librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el data set
dataset = pd.read_csv('Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

from sklearn.impute import SimpleImputer
# Reemplazar por medias
imputer = SimpleImputer(strategy="mean")
#medias en columnas 1,2
imputer = imputer.fit(x[:, 1:3])
# Cambiar valores por dichas nedias
x[:, 1:3] = imputer.transform(x[:, 1:3])

from sklearn.preprocessing import LabelEncoder, OneHotEncoder
from sklearn.compose import ColumnTransformer

labelEncoderX = LabelEncoder()
x[:, 0] = labelEncoderX.fit_transform(x[:, 0])

ct = ColumnTransformer( [('one_hot_encoder', OneHotEncoder(categories='auto'), [0])], remainder='passthrough')
x = np.array(ct.fit_transform(x), dtype=np.float)

labelEncoderY = LabelEncoder()
y = labelEncoderY.fit_transform(y)

# Dividir el dataset en conjunto de entranamiento y testing
from sklearn.model_selection import train_test_split
xTrain, xTest, yTrain, yTest = train_test_split(x, y, test_size = 0.2, random_state = 0)

# Escalar los datos, para tener el mimo ranfod e valores. La distancia de Euclides
from sklearn.preprocessing import StandardScaler
xScaler = StandardScaler()
xTrain = xScaler.fit_transform(xTrain)
xTest = xScaler.transform(xTest)

File: datasets/Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/data_preprocessing.R
#Importar el data set
dataset = read.csv('Data.csv')

#Tratamiento de los dataset
dataset$Age = 
  ifelse(
    is.na(dataset$Age), 
    ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
    dataset$Age)

dataset$Salary = 
  ifelse(
    is.na(dataset$Salary), 
    ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
    dataset$Salary)

# Codificar las variables categoricas
dataset$Country = 
  factor(
    dataset$Country,
    levels = c("France", "Spain", "Germany"),
    labels = c(1, 3, 3))

dataset$Purchased = 
  factor(
    dataset$Purchased,
    levels = c("No", "Yes"),
    labels = c(0, 1))

# Dividir los datos en conjunto de entrenamiento y test
#install.packages("caTools")
library(caTools)
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
trainingSet = subset(dataset, split == TRUE)
testingSet = subset(dataset, split == FALSE)

# Escalar los datos, para tener el mimo ranfod e valores. La distancia de Euclides
trainingSet[, 2:3] = scale(trainingSet[, 2:3])
testingSet[, 2:3] = scale(testingSet[, 2:3])
